engine/consensus/compliance/engine.go

```diff
@@ -303,31 +300,29 @@ func (e *Engine) Process(channel network.Channel, originID flow.Identifier, even
 	return nil
 }
 
-// loop implements the processing of inbound messages. Only returns when Engine is terminated.
-func (e *Engine) loop() {
+// processMessagesLoop processes available block, vote, and timeout messages as they are queued.
+func (e *Engine) processMessagesLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
 	for {
 		select {
-		case <-e.unit.Quit():
+		case <-ctx.Done():
 			return
 		case <-e.messageHandler.GetNotifier():
+			// TODO expected errors?
 			err := e.processAvailableMessages()
 			if err != nil {
-				e.log.Fatal().Err(err).Msg("internal error processing queued message")
+				ctx.Throw(err)
 			}
 		}
 	}
 }
 
-// processAvailableMessages processes any available messages from the inbound queues.
+// processAvailableMessages processes any available messages until the message queue is empty.
 // Only returns when all inbound queues are empty (or the engine is terminated).
-// No errors expected during normal operations.
+// TODO error docs
```

Comment:
has something changed? Previously all sentinels were handled in this function, any error returned from this function is an exception and not expected in normal ops.

Reply:
I changed this because errors are not properly documented or handled yet in calls to `compliance.Core`. The way I have been updating these is to only update error docs for a function once its dependencies are also updated.

Reply:
created PR https://github.com/onflow/flow-go/pull/3243 (targeting this branch) that adds the missing error documentation.